src/Qir/Runtime/lib/QIR/arrays.cpp

```diff
@@ -118,7 +118,7 @@ QirArray::QirArray(TItemCount countItems, TItemSize itemSizeBytes, TDimCount dim
 
     assert(this->count * (TBufSize)itemSizeInBytes < std::numeric_limits<TBufSize>::max());
     // Using `<` rather than `<=` to calm down the compiler on 32-bit arch.
-    const TBufSize bufferSize = this->count * itemSizeInBytes;
+    const TBufSize bufferSize = (TBufSize)this->count * (TBufSize)itemSizeInBytes;

```

Comment:
(nit) For the future: It is probably more efficient to do the operation first, and then cast the result.
 ```suggestion
    const TBufSize bufferSize = (TBufSize)(this->count * itemSizeInBytes);
```

Reply:
I think that would work here, yes. But I was a little concerned about a future where these two types would differ. Right now, `this->count` is of type `TItemCount` while `itemSizeInBytes` is of type `TItemSize`. It just so happens that both of these are `uint32_t`, but since they are independently defined they could vary. If we want to tie these values together more closely, we could consider consolidating `TItemCount` and `TItemSize` into one type.